lib/vehicle_factory_spec.rb

```diff
@@ -0,0 +1,10 @@
+require './lib/vehicle'
+require './lib/vehicle_factory'
+require './lib/dmv_data_service'
+
+Rspec.describe Vehicle_factory do
+    before(:each) do
+        @factory = VehicleFactory.new
+        @wa_ev_registrations = DmvDataService.new.wa_ev_registrations
```

Comment:
Once you `get` this data, a factory class can help you map over the collection and create class instances